tests/lib/rules/no-typos.js

```diff
@@ -1369,6 +1369,17 @@ ruleTester.run('no-typos', rule, {
     }, {
       message: 'Typo in declared prop type: objectof'
     }]
+  }, {
+    code: `
+      class Hello extends React.Component {
+        getDerivedStateFromProps() { }
```

Comment:
let's please add a "valid" rule for `static getDerivedStateFromProps`, and also an invalid one for `static getDERIVEDStateFromProps` :-)

Reply:
I added a test to check that `static getDerivedStateFromProps` is valid.

Regarding the second, there is already a test that checks `static GetDerivedStateFromProps` is invalid. Is it necessary to check `static getDERIVEDStateFromProps` too?

Reply:
as long as the checks cover a) the proper spelling and capitalization, b) any miscapitalization, c) any misspelling, and then those three with 1) static and 2) instance, then we should be good.